packages/semver/src/schematics/ng-add/utils/dependencies.ts

```diff
@@ -0,0 +1,113 @@
+import { noop, Rule, Tree } from '@angular-devkit/schematics';
+import { addDepsToPackageJson, updateJsonInTree } from '@nrwl/workspace';
+import { constants, mkdirSync, writeFileSync } from 'fs';
+
+import { SchemaOptions } from '../schema';
+
+function getExecutableMode() {
+  return constants.S_IXUSR | constants.S_IXGRP | constants.S_IXOTH;

```

Comment:
I think we need `S_IRWXU` instead and we can directly use the constant instead of a function, [here](https://nodejs.org/api/fs.html#fs_file_mode_constants) is the list of fs constants. I would also suggest adding a comment to point that we can't use `Tree.create` for managing permissions and emphasize it's a hack.

Reply:
done